### DISCUSSION TITLE
Why WinPython is getting bigger ... and what to do about it

### DISCUSSION THREAD

**Discussion by stonebig** [winpython/winpython#965] [Q&A]
Looking at current situation, in Winpython64-3.9.4.0rc of 2021-02 of 747 Mo, you have:
- numpy+mkl is 220 Mo 
- Torch-cpu is 185 Mo
- Qt is roughly 100 Mo
- you don't have VSCode, which is .. 240 Mo

Perceived Actions, only indirect:
- push Torch in hope Microsoft will ship the "Fat" itself in Windows 10,
- push new C interfaces of Python / HPY / pypy to reduce needs for package to ship their binary version of numpy or MKL or Chromium,
- pushing Qt6 in hope it pushes Web convergence, and so drop some redundancy.

**Comment by RoyiAvital** [comment]
Why are you troubled with this?
I'd even say you should add more and more packages. Not to be troubled by the download size.

By the way, why do you use Torch+CPU? There is no point in having no GPU support.

**Reply by stonebig** [reply]
When gpu support was only NVidia, there was no point of shipping a gpu package, and there is no Nvidia card in a classroom. That may change when a solution works with Intel or AMD integrated ML_processors/gpu or windows intemediation layer.

**Reply by RoyiAvital** [reply]
It makes sense since 90% of those who use GPU for DL (And for something modern there is no other choice for the end user) know they should buy nVidia.

In case they don't, it will work on the CPU as the current package.
